src/main.rs

```diff
@@ -31,6 +31,20 @@ struct EvgExpansions {
     pub task_name: String,
     /// ID of Evergreen version running.
     pub version_id: String,
+    /// True if the patch is a patch build.
+    #[serde(default, deserialize_with = "deserialize_is_patch")]
+    pub is_patch: bool,
+}
+
+fn deserialize_is_patch<'de, D>(deserializer: D) -> Result<bool, D::Error>
```

Comment:
I had to add this because the `is_patch` variable is unfortunately not a boolean, but a string. This is to deserialize that string into a boolean.

Reply:
From [evg docs](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Configuration-Files#default-expansions)
> ${is_patch} is "true" if the running task is in a patch build and undefined if it is not.

Is it going to work as expected outside of patch buids?

Reply:
I've added a [`#[serde(default)]` above](https://github.com/mongodb/mongo-task-generator/pull/83/files/3028104ee63e3d6d37a55be8fb4b435cc1bac789#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR35) - this is so that if the `is_patch` variable doesn't exist in the YAML altogether, then it defaults to `false` when it is deserialized.

The way I tested the "missing `is_patch` case" is by changing [this code](https://github.com/mongodb/mongo-task-generator/pull/83/files/3028104ee63e3d6d37a55be8fb4b435cc1bac789#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR36) to attempt to deserialize an `is_patch_without` variable, which will not exist in the Evg expansions file. You can see that the mongo-task-generator was still able to run fine in this [patch](https://spruce.mongodb.com/version/670990beaec3160007a92e56/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) but it omitted the `--useMinimalTests` option.

If the variable exists, and is set to anything other than the string `"true"`, then again it becomes false. This is done [here](https://github.com/mongodb/mongo-task-generator/pull/83/files/3028104ee63e3d6d37a55be8fb4b435cc1bac789#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR46). Only when it is set to `"true"` is it actually deserialized into the boolean with `true`. I tested this and the above as well locally, but I couldn't figure out a way to test this case on Evg.

Reply:
Thanks for more detailed explanation. Even though I did some things in Rust for this project, it's still hard to comprehend for me sometimes.